sway-core/src/semantic_analysis/namespace/trait_map.rs

```diff
@@ -1065,7 +1073,7 @@ impl TraitMap {
             })
             .collect();
 
-        let relevant_impld_traits: BTreeMap<Ident, TypeId> = all_impld_traits
+        let relevant_impld_traits: BTreeSet<(Ident, TypeId)> = all_impld_traits
```

Comment:
This is a question for the line 1052 actually, that was not changed. There we still have the `BTreeMap<Ident, TypeId>` when collecting constraints. Isn't it that with the constraints we can also have the situation of the same trait and different types e.g. `where T: Trait<u64> + Trait<u32>`? Shouldn't that also be `BTreeSet` like the other variables?